modules/createHistory.js

```diff
@@ -69,21 +69,21 @@ function createHistory(options={}) {
   function getTransitionConfirmationMessage() {
     var message = null;
 
-    for (var i = 0, len = transitionHooks.length; i < len && typeof message !== 'string'; ++i)
+    for (var i = 0, len = transitionHooks.length; i < len && message === null; ++i)
       message = transitionHooks[i].call(this);
 
     return message;
   }
 
   function confirmTransition(callback) {
-    var message;
+    var message = getTransitionConfirmationMessage();
 
-    if (getUserConfirmation && (message = getTransitionConfirmationMessage())) {
+    if (getUserConfirmation && message) {
```

Comment:
Let's use `typeof message === 'string'` here so we don't pass non-strings to `getUserConfirmation`.